flyteplugins/go/tasks/pluginmachinery/k8s/client.go

```diff
@@ -40,9 +38,14 @@ type Options struct {
 // NewKubeClient creates a new KubeClient that caches reads and falls back to
 // make API calls on failure. Write calls are not cached.
 func NewKubeClient(config *rest.Config, options Options) (core.KubeClient, error) {
+	httpClient, err := rest.HTTPClientFor(config)
```

Comment:
A question. We used to have three different http clients for different purposes, and now we will only have one. Do you know the implication, or do you prefer we still have three? @eapolinario

Reply:
To be clear, I don't see why we would need three different struct instances given the config is the same.

Reply:
```
// The Client's Transport typically has internal state (cached TCP
// connections), so Clients should be reused instead of created as
// needed. Clients are safe for concurrent use by multiple goroutines.
```

Reply:
I think this was the PR https://github.com/kubernetes-sigs/controller-runtime/pull/2122, which carried exactly the same rationale.